swiftsimio/visualisation/slice.py

```diff
@@ -396,13 +401,18 @@ def slice_gas_pixel_grid(
     else:
         image = slice_scatter(**common_parameters)
 
-    return image
+    # determine the effective number of pixels for each dimension
+    xres = int(resolution * x_range / max_range)
+    yres = int(resolution * y_range / max_range)
+
+    # trim the image to remove empty pixels
+    return image[:xres, :yres]
 
 
 def slice_gas(
     data: SWIFTDataset,
     resolution: int,
-    slice: float,
+    z_slice: unyt_quantity = 0.0 * unyt.m,
```

Comment:
I don't like this - can we have `z_slice: Optional[unyt_quantity] = None` and set it to `0.0 * data.gas.coordinates.units`?